fabric-input-events-v1/src/main/java/net/fabricmc/fabric/api/event/client/input/GenericKeyEvent.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.client.input;
+
+import net.minecraft.client.util.InputUtil.Key;
+
+public abstract class GenericKeyEvent {
```

Comment:
Instead of allocating a new object, just pass these objects directly.

Reply:
That's true right now, but [there are several ways around that](https://github.com/FabricMC/fabric/pull/1272#discussion_r559790013).